### ISSUE TITLE
Error when renaming a file when @types exists

### ISSUE THREAD

**Issue by gregjacobs** [dsherret/ts-simple-ast#405]
Hey @dsherret. Looks like another interesting one for you. Getting an error when renaming a file. Check out this reproduction:

```ts
// test.ts
import Project from 'ts-simple-ast';

const project = new Project();

const sourceFile1 = project.createSourceFile( 'source-file-1.js', `
	import $ from 'jquery';
` );
const sourceFile2 = project.createSourceFile( 'source-file-2.js', `
	// no content
` );

// rename from js to ts
sourceFile1.move( 'source-file-1.ts' );
sourceFile2.move( 'source-file-2.ts' );
```

This script on its own does not produce an error. However, when the `@types/jquery` package is installed, it produces an error on the *second* file rename.

```bash
$ npm install @types/jquery    # oh noes :( 

$ ts-node test.ts
/Users/gjacobs/dev/js-to-ts-converter/node_modules/ts-simple-ast/dist/errors/BaseError.js:7
        var _this = _super.call(this, message) || this;
                           ^
Error: Attempted to get information from a node that was removed or forgotten.
    at InvalidOperationError.BaseError [as constructor] (/Users/gjacobs/dev/js-to-ts-converter/node_modules/ts-simple-ast/dist/errors/BaseError.js:7:28)
    at new InvalidOperationError (/Users/gjacobs/dev/js-to-ts-converter/node_modules/ts-simple-ast/dist/errors/InvalidOperationError.js:8:28)
    at NamespaceDeclaration.get [as compilerNode] (/Users/gjacobs/dev/js-to-ts-converter/node_modules/ts-simple-ast/dist/compiler/common/Node.js:31:23)
    at NamespaceDeclaration.Node.getKind (/Users/gjacobs/dev/js-to-ts-converter/node_modules/ts-simple-ast/dist/compiler/common/Node.js:106:21)
    at Function.ModuleUtils.getReferencedSourceFileFromSymbol (/Users/gjacobs/dev/js-to-ts-converter/node_modules/ts-simple-ast/dist/utils/compiler/ModuleUtils.js:14:58)
    at ImportDeclaration.getModuleSpecifierSourceFile (/Users/gjacobs/dev/js-to-ts-converter/node_modules/ts-simple-ast/dist/compiler/file/ImportDeclaration.js:47:36)
    at SourceFileReferenceContainer.getSourceFileForLiteral (/Users/gjacobs/dev/js-to-ts-converter/node_modules/ts-simple-ast/dist/utils/references/SourceFileReferenceContainer.js:148:27)
    at SourceFileReferenceContainer.resolveUnresolved (/Users/gjacobs/dev/js-to-ts-converter/node_modules/ts-simple-ast/dist/utils/references/SourceFileReferenceContainer.js:17:40)
    at EventContainer.fire (/Users/gjacobs/dev/js-to-ts-converter/node_modules/ts-simple-ast/dist/utils/EventContainer.js:37:17)
    at CompilerFactory.replaceCompilerNode (/Users/gjacobs/dev/js-to-ts-converter/node_modules/ts-simple-ast/dist/factories/CompilerFactory.js:479:48)
```

This is happening in v14.2.0, but actually didn't happen in v12.x. 

Any ideas? :X :)

**Comment by gregjacobs** [comment]
Btw this one took me quite a while to track down :) Was a stroke of luck that I figured out it was the typings!

**Comment by dsherret** [comment]
Crazy! It's because of the memoizing I added on symbol. That must mean symbols are mutated by the compiler.

Thanks a lot for the reproduction steps and your comment about it only recently happening allowed me to quickly find the problem.

Fixed in 14.2.1.

**Comment by gregjacobs** [comment]
Very interesting! It's too bad that the compiler mutates things behind the scenes. Wish it followed more of an immutable pattern.. Oh well!

Thanks for the quick fix!
